cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -441,6 +441,10 @@ private Object convertValue(TypeToken<?> fieldType, String value) throws Excepti
         }
       }
 
+      if (Long.class.equals(rawType)) {
+        value = value.substring(0, value.lastIndexOf("."));
```

Comment:
This is too hardcode. What if the serialization changed again and it doesn't have "." anymore?